test/e2e/traceflow_test.go

```diff
@@ -0,0 +1,321 @@
+						{
+							Component:     v1alpha1.Forwarding,
+							ComponentInfo: "Output",
+							Action:        v1alpha1.Forwarded,
+						},
+					},
+				},
+				{
+					Node: node2,
+					Observations: []v1alpha1.Observation{
+						{
+							Component:     v1alpha1.Forwarding,
+							ComponentInfo: "Classification",
+							Action:        v1alpha1.Received,
+						},
+						{
+							Component:     v1alpha1.Forwarding,
+							ComponentInfo: "Output",
+							Action:        v1alpha1.Delivered,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Setup 2 NetworkPolicies:
+	// 1. Allow all egress traffic.
+	// 2. Deny ingress traffic on pod with label antrea-e2e = node1Pods[1]. So flow node1Pods[0] -> node1Pods[1] will be dropped.
+	var allowAllEgress *networkingv1.NetworkPolicy
+	if allowAllEgress, err = data.createNPAllowAllEgress("test-networkpolicy-allow-all-egress"); err != nil {
+		t.Fatalf("Error when creating network policy: %v", err)
+	}
+	defer func() {
+		if err = data.deleteNetworkpolicy(allowAllEgress); err != nil {
+			t.Fatalf("Error when deleting network policy: %v", err)
+		}
+	}()
+
+	var denyIngress *networkingv1.NetworkPolicy
+	if denyIngress, err = data.createNPDenyAllIngress("antrea-e2e", node1Pods[1], "test-networkpolicy-deny-ingress"); err != nil {
+		t.Fatalf("Error when creating network policy: %v", err)
+	}
+	defer func() {
+		if err = data.deleteNetworkpolicy(denyIngress); err != nil {
+			t.Fatalf("Error when deleting network policy: %v", err)
+		}
+	}()
+
+	// Create 2 traceflows:
+	// 1. node1Pods[0] -> node1Pods[1], intra node1.
+	// 2. node1Pods[0] -> node2Pods[0], inter node1 and node2.
+	if intraNodeTraceflow, err = data.crdClient.OpsV1alpha1().Traceflows().Create(context.TODO(), intraNodeTraceflow, v1.CreateOptions{}); err != nil {
+		t.Fatalf("Error when creating traceflow: %v", err)
+	}
+	defer func() {
+		if err := data.crdClient.OpsV1alpha1().Traceflows().Delete(context.TODO(), intraNodeTraceflow.Name, v1.DeleteOptions{}); err != nil {
+			t.Fatalf("Error when deleting traceflow: %v", err)
+		}
+	}()
+
+	if interNodeTraceflow, err = data.crdClient.OpsV1alpha1().Traceflows().Create(context.TODO(), interNodeTraceflow, v1.CreateOptions{}); err != nil {
+		t.Fatalf("Error when creating traceflow: %v", err)
+	}
+	defer func() {
+		if err := data.crdClient.OpsV1alpha1().Traceflows().Delete(context.TODO(), interNodeTraceflow.Name, v1.DeleteOptions{}); err != nil {
+			t.Fatalf("Error when deleting traceflow: %v", err)
+		}
+	}()
+
+	// Wait for traceflows to finish.
+	if err = wait.Poll(1*time.Second, traceflowTimeout, func() (bool, error) {
+		done := 0
+		for i := 0; i < len(testcases); i++ {
+			tc := testcases[i]
+			tf := tc.tf
+			if tf, err = data.crdClient.OpsV1alpha1().Traceflows().Get(context.TODO(), tf.Name, v1.GetOptions{}); err != nil || tf.Status.Phase != tc.expectedPhase {
+				return false, nil
+			} else {
+				if tf.Status.Results == nil || (len(tf.Status.Results) != len(tc.expectedResults)) {
```

Comment:
try to have less indent, this doesn't need to intent as the above condition returns.